### PR TITLE
Pixel shader upgrading

### PR DESCRIPTION
**Pull request by PatrickvL**
A minor bit of maintenance on our pixel shader generation code;
* Removed support for shader models older than 2.0 while keeping existing functionality intact.
* Fixed combiner output flags parsing (it didn't use a bitmask before)
* Separate decoding of register combiner registers from generation of pixel shaders.

Only the second might solve some corner-cases, the others won't cause any functional change.

### CODE REVIEW

File: src/core/hle/D3D8/XbPixelShader.cpp [hunk 1]
```diff
@@ -4270,24 +4208,18 @@ bool PSH_XBOX_SHADER::FixArgumentModifiers()
...
 			continue;
 		
-        if (m_PSVersion < D3DPS_VERSION(2, 0) && Cur->Opcode == PO_MOV)
-            continue;
-
         int InsertPos = i;
         // Detect modifiers on constant and arguments
 		for (int p = 0; p < 7 && p < PSH_OPCODE_DEFS[Cur->Opcode]._In; p++) {
-			if ((Cur->Parameters[p].Type == PARAM_C || (m_PSVersion >= D3DPS_VERSION(2, 0) && Cur->Parameters[p].UsesRegister()))
+			if ((Cur->Parameters[p].Type == PARAM_C || (Cur->Parameters[p].UsesRegister()))
```
Comment (RadWolfie) [inline src/core/hle/D3D8/XbPixelShader.cpp]: Found extra round brackets for `Cur->Parameters[p].UsesRegister()`.

**Comment by RadWolfie** [comment]
Beside one thing I commented above, the code looks good to me and I had not discover any regression in most titles I had tested.

**Comment by LukeUsher** [comment]
Any test results for any improvements and/or regressions? 
It would be nice to see if there have been any changes as a result of this work

**Comment by PatrickvL** [comment]
The cases that previously weren't hit, were: PS_COMBINEROUTPUT_SHIFTLEFT_1_BIAS, PS_COMBINEROUTPUT_SHIFTLEFT_2=         0x20L, and PS_COMBINEROUTPUT_SHIFTRIGHT_1

**Comment by LukeUsher** [comment]
Once the conflict is resolved, this can be merged

**Comment by PatrickvL** [comment]
Status : Rebased and resolved conflict, removed needless round brackets and added LOG_TEST_CASE calls for the cases that weren't hit before.

**Comment by RadWolfie** [comment]
Hmm... something went wrong, the whole file has been changed during the rebase. Which is caused by commit 3587deb.

**Comment by RadWolfie** [comment]
So far, no issue with few titles tested.

One thing I had found is this check exist. Haven't found other checks below 2.0.
https://github.com/Cxbx-Reloaded/Cxbx-Reloaded/blob/54daf75a1cb3ed027e3e7225862339f635f1ed63/src/core/hle/D3D8/XbPixelShader.cpp#L5580

**Comment by PatrickvL** [comment]
Ready for final review and merge

**Comment by RadWolfie** [comment]
Tested with at least 10 titles, I had not seen any regressions after the change to use IsArithetic function.

**Comment by LukeUsher** [comment]
Yeah this looks fairly low risk, I'm quite happy to merge

**Pull request merged**